### ISSUE TITLE
Use dependencies from `_build/dependencies` folder

### ISSUE THREAD

**Issue by peq** [wurstscript/WurstScript#562]
- [ ] Teach compiler to find dependencies in `_build/dependencies` automatically (folder created by setup tool)
- [ ] Make editor track changes to files in `_build/dependencies` (basically handle them like files in the `wurst` folder)
- [ ] Before compiling clean packages not from `wurst` folder (i.e. packages from dependencies`)

This has the following benefits:
- updates in dependencies are directly reflected in the overall project (no more clean necessary)
- Import autocomplete and quickfixes know about all sources and thus are more useful (no more lazy loading of dependencies)

**Comment by peq** [comment]
I think the first two points already work as expected. So only the third point (cleaning unused packages from dependencies) needs to be implemented. Or am I missing something?

**Comment by Frotty** [comment]
No, the compiler doesn't find them "automatically". It's still done via the setup generating a Wurst.dependencies file, which points to the folders inside the dependencies folder.
Iirc we wanted to get rid of that dependencies file.
The track changes is also still broken I think. Changing Dependency packages requires clean or can cause ambiguity errors. I will recheck on this.

**Comment by peq** [comment]
I added some code to automatically add folders from `_build/dependencies` in 3a3e1b71907aaf80e937f4dac9c8aef1fca00efd

Can you check if that works as intended for you? I'll also try to do some unit tests for this.
